system-test/read-rows.ts

```diff
@@ -12,69 +12,76 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import {Bigtable} from '../src';
-import {Mutation} from '../src/mutation.js';
+import {Bigtable, protos, Table} from '../src';
 const {tests} = require('../../system-test/data/read-rows-retry-test.json') as {
-  tests: Test[];
+  tests: ReadRowsTest[];
 };
 import {google} from '../protos/protos';
 import * as assert from 'assert';
-import {describe, it, afterEach, beforeEach} from 'mocha';
-import * as sinon from 'sinon';
-import {EventEmitter} from 'events';
-import {Test} from './testTypes';
-import {ServiceError, GrpcClient, GoogleError, CallOptions} from 'google-gax';
-import {PassThrough} from 'stream';
+import {describe, it, before} from 'mocha';
+import {ReadRowsTest} from './testTypes';
+import {ServiceError, GrpcClient, CallOptions} from 'google-gax';
+import {MockServer} from '../src/util/mock-servers/mock-server';
+import {MockService} from '../src/util/mock-servers/mock-service';
+import {BigtableClientMockService} from '../src/util/mock-servers/service-implementations/bigtable-client-mock-service';
+import {ServerWritableStream} from '@grpc/grpc-js';
 
 const {grpc} = new GrpcClient();
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-function dispatch(emitter: EventEmitter, response: any) {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const emits: any[] = [{name: 'request'}];
-  if (response.row_keys) {
-    emits.push.apply(emits, [
-      {name: 'response', arg: 200},
-      {
-        name: 'data',
-        arg: {chunks: response.row_keys.map(rowResponse)},
-      },
-    ]);
-  }
-  if (response.end_with_error) {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const error: any = new Error();
-    error.code = response.end_with_error;
-    emits.push({name: 'error', arg: error});
-  } else {
-    emits.push({name: 'end'});
-  }
-  let index = 0;
-  setImmediate(next);
-
-  function next() {
-    if (index < emits.length) {
-      const emit = emits[index];
-      index++;
-      emitter.emit(emit.name, emit.arg);
-      setImmediate(next);
-    }
-  }
-}
-
-function rowResponse(rowKey: {}) {
+function rowResponseFromServer(rowKey: string) {
   return {
-    rowKey: Mutation.convertToBytes(rowKey),
+    rowKey: Buffer.from(rowKey).toString('base64'),
     familyName: {value: 'family'},
-    qualifier: {value: 'qualifier'},
-    valueSize: 0,
-    timestampMicros: 0,
-    labels: [],
+    qualifier: {value: Buffer.from('qualifier').toString('base64')},
     commitRow: true,
-    value: 'value',
+    value: Buffer.from(rowKey).toString('base64'),
   };
 }
 
+function getRequestOptions(request: any): google.bigtable.v2.IRowSet {
```

Comment:
From the old test, this contains all the code which was used for building a request.